marklogic-data-hub/src/main/resources/ml-modules/root/data-hub/5/impl/entity-lib.sjs

```diff
@@ -208,6 +209,97 @@ function deleteModel(entityName) {
   }
 }
 
+/**
+ * Returns the step names that contain a reference to the supplied entity.
+ * The targetEntityType for mapping artifacts is checked against both an entityName or an entityTypeId.
+ *
+ * @param entityName
+ * @param entityTypeId
+ * @returns {[]}
+ */
+function findModelReferencesInSteps(entityName, entityTypeId) {
+  const steps = [];
+  const mappingArtifacts = cts.search(cts.andQuery([
+    cts.collectionQuery('http://marklogic.com/data-hub/mappings'),
+    cts.jsonPropertyValueQuery("targetEntityType", [entityName, entityTypeId])
+  ]));
+  mappingArtifacts.toObject()
+    .forEach(mapping => {
+      mapping = JSON.parse(mapping);
```

Comment:
I think you just need mapping.toObject() here? Could get fancy and do this (I think this works - I also often like to declare a query in a separate variable so it's easy to understand outside of the context where it's used, and the variable name can describe the intent of the query):

```
const mappingQuery = ...
const steps = cts.search(mappingQuery).toArray().map(mapping => mapping.toObject().name);
```